### PR TITLE
Missing xml-model reference generates multiple similar warnings

### LINKED ISSUE #795: Missing xml-model reference generates multiple similar warnings
Try to open this (file.xsd is obviously missing)
```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="file.xsd" type="application/xml" schematypens="http://www.w3.org/2001/XMLSchema"?>
<classpath>
	<classpathentry kind="src" output="bin/main" path="src/main/java">
		<attributes>
			<attribute name="gradle_scope" value="main"/>
			<attribute name="gradle_used_by_scope" value="main,test"/>
		</attributes>
	</classpathentry>
	<classpathentry kind="src" output="bin/main" path="src/main/resources">
		<attributes>
			<attribute name="gradle_scope" value="main"/>
			<attribute name="gradle_used_by_scope" value="main,test"/>
		</attributes>
	</classpathentry>
	<classpathentry kind="src" output="bin/test" path="src/test/java">
		<attributes>
			<attribute name="gradle_scope" value="test"/>
			<attribute name="gradle_used_by_scope" value="test"/>
			<attribute name="test" value="true"/>
		</attributes>
	</classpathentry>
	<classpathentry kind="src" output="bin/test" path="src/test/resources">
		<attributes>
			<attribute name="gradle_scope" value="test"/>
			<attribute name="gradle_used_by_scope" value="test"/>
			<attribute name="test" value="true"/>
		</attributes>
	</classpathentry>
	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8/"/>
	<classpathentry kind="con" path="org.eclipse.buildship.core.gradleclasspathcontainer"/>
	<classpathentry kind="output" path="bin/default"/>
</classpath>
```

See the multitude of `schema_reference.4: Failed to read schema document 'untitled:file.xsd', because 1) could not find the document; 2) the document could not be read; 3) the root element of the document is not <xsd:schema>.` warnings:

Screen Shot 2020-06-22 at 5 54 08 PM

### PR DESCRIPTION
**Pull request by angelozerr**
Missing xml-model reference generates multiple similar warnings

Fixes #795

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
@fbricon the draft PR should fix the issue https://github.com/eclipse/lemminx/issues/795 and fix the issue with DTD too.

The fix gives the capability to provide code action to generate the XSD or DTD. 

I need to clean my code and write tests to swith to real PR.

**Comment by angelozerr** [comment]
My PR is now clean and tests are written.

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/XMLModelUtils.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+		}
+		return null;
+	}
+
+	/**
+	 * Returns the expanded system location
+	 *
+	 * @return the expanded system location
+	 */
+	private static String getResolvedLocation(String documentURI, String location) {
```
[the remark this answers is not in the document]
Comment (angelozerr) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/XMLModelUtils.java:61]: Indeed we should provide that in a utility class like URIUtils. The main problem is that thid method depends on Xerces. I would like to provide URIUtils#getResolvedLocation without Xerces dependencies. In other words I think it should be better to do that in an another issue.

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/xerces/xmlmodel/msg/XMLModelMessageFormatter.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        catch (MissingResourceException e) {
+            msg = fResourceBundle.getString("BadMessageKey");
+            throw new MissingResourceException(key, msg, key);
+        }
+
+        // no message
+        if (msg == null) {
+            msg = key;
+            if (arguments.length > 0) {
+                StringBuffer str = new StringBuffer(msg);
```
[the remark this answers is not in the document]
Comment (angelozerr) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/xerces/xmlmodel/msg/XMLModelMessageFormatter.java]: fixed

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/XMLModelUtils.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (angelozerr) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/XMLModelUtils.java:61]: I create an issue for that https://github.com/eclipse/lemminx/issues/801

**Comment by xorye** [comment]
Not completely related to this PR, but I think for this part:
https://github.com/eclipse/lemminx/blob/15fd1d6034bf36946c08fda49a5fd93048798287/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/XMLSchemaErrorCode.java#L198-L208

could we use
https://github.com/eclipse/lemminx/blob/d102a4c637f5372ed2f015d73adde82fee066e5c/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/utils/XMLPositionUtility.java#L914-L921

instead?

**Comment by angelozerr** [comment]
> Not completely related to this PR, but I think for this part:

Good catch! Fixed.

**Pull request merged**